### PR TITLE
invoke the conformal model (WIP)

### PR DESCRIPTION
**Pull request by benblamey**

### CODE REVIEW

File: haste/windowed_conformal_model/time_series_features.py [hunk 1]
```diff
@@ -14,6 +14,8 @@
 import pandas as pd
 from pygam import LinearGAM
 
+
+# TODO: reorder params, timestamps first
 def time_series_features(feature_time_series, timestamps, end_time):
```
Comment (benblamey) [inline haste/windowed_conformal_model/time_series_features.py:19]: OK, so we compute the ts-features on the whole well-substream, in which case, what are the values for `all_features`, `features_for_new_image` we pass into the conformal model from here?

File: haste/windowed_conformal_model/conformal_interestingness_model.py [hunk 2]
```diff
@@ -91,27 +95,41 @@ def interestingness(self,
...
+            sum_of_intensities_ts_features = time_series_features(sum_of_intensities_ndarray,
+                                                                  timestamps_ndarray,
+                                                                  timestamp)
+            # = (mean,sd,d1,d2)
 
-            # TODO: interpret the conformal prediction - make a binary decision 1 or 0 on interestingness.
-            # based on epsilon
+            # TODO: query the core model (pass in whatever it needs, also window size)
+            # BB: again, confused about the windowing?!
+            pval = 0.1  # interestingness(all_features, features_for_new_image, all_y):
```
Comment (benblamey) [inline haste/windowed_conformal_model/conformal_interestingness_model.py]: confused here.

File: haste/windowed_conformal_model/conformal_interestingness_model.py [hunk 3]
```diff
@@ -91,27 +95,41 @@ def interestingness(self,
...
+            sum_of_intensities = course_features[1]
+            sum_of_intensities_ndarray = numpy.ndarray(shape=(len(sum_of_intensities)),
+                                                       dtype=int,
+                                                       buffer=numpy.array(sum_of_intensities))
 
-            # TODO: query the core model (pass in whatever it needs, also window size)
+            # Compute features on the entire timeseries for that well.
+            sum_of_intensities_ts_features = time_series_features(sum_of_intensities_ndarray,
+                                                                  timestamps_ndarray,
+                                                                  timestamp)
```
Comment (benblamey) [inline haste/windowed_conformal_model/conformal_interestingness_model.py]: computing TS features for all-time

File: haste/windowed_conformal_model/time_series_features.py (hunk 1 above)
Comment (philharr) [reply haste/windowed_conformal_model/time_series_features.py:19]: This function returns the features_for_new_image. The function will have already been used offline to compute all_features for the "training" data. These are then fed into the interestingness function

File: haste/windowed_conformal_model/conformal_interestingness_model.py (hunk 2 above)
Comment (philharr) [reply haste/windowed_conformal_model/conformal_interestingness_model.py]: interestingness will return two p-values. One for the 0 category (not interesting) and one for the 1 category (interesting).

File: haste/windowed_conformal_model/conformal_interestingness_model.py (hunk 3 above)
Comment (philharr) [reply haste/windowed_conformal_model/conformal_interestingness_model.py]: Ok

**Review by philharr — COMMENTED** [review]
Hope things clarified

File: haste/windowed_conformal_model/conformal_interestingness_model.py [hunk 4]
```diff
@@ -91,27 +95,44 @@ def interestingness(self,
...
-            # based on epsilon
+            p_interesting = p_values[0]
+            p_not_interesting = p_values[1]
 
-            # TODO: return that value
-            return {'interestingness': 1}
+            if p_not_interesting > EPSILON:
+                return {'interestingness': 0}
+            else:
+                return {'interestingness': 1}
```
Comment (benblamey) [inline haste/windowed_conformal_model/conformal_interestingness_model.py]: is this correct? small p-values means more certainty?

File: haste/windowed_conformal_model/conformal_interestingness_model.py [hunk 5]
```diff
@@ -91,27 +95,44 @@ def interestingness(self,
...
-            # based on epsilon
+            p_interesting = p_values[0]
+            p_not_interesting = p_values[1]
 
-            # TODO: return that value
-            return {'interestingness': 1}
+            if p_not_interesting < EPSILON:
+                return {'interestingness': 0}
+            else:
+                return {'interestingness': 1}
```
Comment (benblamey) [inline haste/windowed_conformal_model/conformal_interestingness_model.py]: is this correct? small values mean more certainty of the classification?

File: haste/windowed_conformal_model/conformal_interestingness_model.py [hunk 6]
```diff
@@ -91,27 +96,43 @@ def interestingness(self,
...
+                                               buffer=numpy.array(timestamps))
 
+            # TODO: which image features do we want to use?
+            sum_of_intensities = course_features[1]
+            sum_of_intensities_ndarray = numpy.ndarray(shape=(len(sum_of_intensities)),
+                                                       dtype=int,
+                                                       buffer=numpy.array(sum_of_intensities))
 
+            # Compute features on the entire timeseries for that well.
+            sum_of_intensities_ts_features = time_series_features(sum_of_intensities_ndarray,
```
Comment (benblamey) [inline haste/windowed_conformal_model/conformal_interestingness_model.py:126]: @Hakanwie just using the sum of intensities here - is this correct?

Comment (philharr) [reply haste/windowed_conformal_model/conformal_interestingness_model.py:126]: we should use both the sum and the correlation (4 time series measures each, appended together, and passed as a vector into the interestingness function)

**Comment by benblamey** [comment]
Merging this, want to see if it blows up...

**Pull request merged**